### PR TITLE
Add support for AI/DI/Customized module

### PR DESCRIPTION
**Pull request by huaqianli**
iot2050-eio-manager: Correct the firmware update typo
iot2050-firmware-update: Correct parameter definition
Revert "build(deps): bump next in /recipes-app/iot2050-conf-webui/files"
linux-iot2050: Generate patch correctly
iot2050-conf-webui: Add support for sm1223 dc di16 dq16 rly
iot2050-eio-manager: Add support for sm1223 dc di16 dq16 rly
iot2050-conf-webui: Add support for sm1221 8di module
iot2050-eio-manager: Add support for sm1221 8di module
iot2050-conf-webui: Add support for sm1231 4AI module
iot2050-eio-manager: Add support for sm1231 4AI module
iot2050-conf-webui: Add support for module SM SENS DI
iot2050-eio-manager: Add support for module SM SENS DI
Revert "board-bootup: setup pwm pins during boot"
Revert "mraa: fix non-root access"
Revert "kas: add iot2050 user to the example image"
Revert "mraa: fix pinmux of user pin"

### CODE REVIEW

File: recipes-app/iot2050-conf-webui/files/npm-shrinkwrap.json
Comment (BaochengSu) [inline recipes-app/iot2050-conf-webui/files/npm-shrinkwrap.json:1]: Explain why this need to be reverted in the commit message

File: recipes-bsp/u-boot/files/0013-iot2050-add-smbios-and-sysinfo-support.patch
Comment (BaochengSu) [inline recipes-bsp/u-boot/files/0013-iot2050-add-smbios-and-sysinfo-support.patch:1]: Do not forget to upstream it first.

Or this is the pre-review for the upstreaming? If so, split it into another PR, I guess this would require more time than other parts in this PR.

File: recipes-app/iot2050-conf-webui/files/src/lib/uiString/SM1223.json
Comment (BaochengSu) [inline recipes-app/iot2050-conf-webui/files/src/lib/uiString/SM1223.json:1]: So this is actually a variant of SM1223?

If so, I think there are too much redundant code, please consider folder this into the current SM1223 module.

NAK for this commit.

File: recipes-app/iot2050-conf-webui/files/src/components/ExtendedModules/SM1221_8DI/SM1221with8DIConf.js
Comment (BaochengSu) [inline recipes-app/iot2050-conf-webui/files/src/components/ExtendedModules/SM1221_8DI/SM1221with8DIConf.js:1]: This commit includes both conf-webui and eio-manager, please split them into two different commits.

File: recipes-app/iot2050-conf-webui/files/src/components/ExtendedModules/SM1231_4AI/SM1231with4AIConf.js
Comment (BaochengSu) [inline recipes-app/iot2050-conf-webui/files/src/components/ExtendedModules/SM1231_4AI/SM1231with4AIConf.js:1]: This one as well.

File: recipes-bsp/u-boot/files/0012-smbios-enhance-system-information-retrieval.patch [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
++			t->serial_number = smbios_add_prop_si(ctx, "serial",
++				SYSINFO_ID_SMBIOS_SYSTEM_SERIAL_NUMBER);
+ 	}
++	if (!sysinfo_get_str(ctx->dev, SYSINFO_ID_SMBIOS_SYSTEM_UUID,
++			     sizeof(the_uuid), the_uuid))
++		strncpy((char *)t->uuid, the_uuid, sizeof(t->uuid));
++
+ 	t->sku_number = smbios_add_prop(ctx, "sku");
+ 	t->family = smbios_add_prop(ctx, "family");
+
```
Comment (jan-kiszka) [inline recipes-bsp/u-boot/files/0012-smbios-enhance-system-information-retrieval.patch:53]: This is a must for upstream-first: We can't risk introducing this interface to our board and then being rejected with the generic change by upstream later.

File: recipes-bsp/u-boot/files/0013-iot2050-add-smbios-and-sysinfo-support.patch
Comment (jan-kiszka) [reply recipes-bsp/u-boot/files/0013-iot2050-add-smbios-and-sysinfo-support.patch:1]: Indeed.

File: recipes-bsp/u-boot/files/0013-iot2050-add-smbios-and-sysinfo-support.patch [hunk 2]
```diff
@@ -0,0 +1,530 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Li Hua Qian <huaqian.li@siemens.com>
+Date: Thu, 20 Jun 2024 15:32:37 +0800
+Subject: [PATCH] iot2050: add smbios and sysinfo support
+
```
Comment (jan-kiszka) [inline recipes-bsp/u-boot/files/0013-iot2050-add-smbios-and-sysinfo-support.patch:5]: Missing commit message.

File: recipes-bsp/u-boot/files/0013-iot2050-add-smbios-and-sysinfo-support.patch [hunk 3]
```diff
@@ -0,0 +1,530 @@
...
+-	char mlfb[18 + 1];
+-	char uuid[32 + 1];
+-	char a5e[18 + 1];
+-	u8 mac_addr_cnt;
+-	u8 mac_addr[8][ARP_HLEN];
+-	char seboot_version[40 + 1];
+-	u8 padding[3];
+-	u32 ddr_size_mb;
+-} __packed;
++#include "../../../../drivers/sysinfo/iot2050.h"
```
Comment (jan-kiszka) [inline recipes-bsp/u-boot/files/0013-iot2050-add-smbios-and-sysinfo-support.patch:79]: This looks not so nice. Not sure if this relative include deep into the driver is acceptable. I rather suspect the header needs to move to include/.

File: recipes-bsp/u-boot/files/0013-iot2050-add-smbios-and-sysinfo-support.patch [hunk 4]
```diff
@@ -0,0 +1,530 @@
...
++++ b/common/spl/Kconfig
+@@ -1463,6 +1463,11 @@ config SPL_AT91_MCK_BYPASS
+ 	  The external source has to provide a stable clock on the XIN pin.
+ 	  If this option is disabled, the SoC expects a crystal oscillator
+ 	  that needs driving on both XIN and XOUT lines.
++
++config SPL_SYSINFO
++	bool "Support sysinfo drivers"
++	help
++	  Enable support for sysinfo drivers in SPL.
```
Comment (jan-kiszka) [inline recipes-bsp/u-boot/files/0013-iot2050-add-smbios-and-sysinfo-support.patch:304]: This likely needs to factored out into a separate commit, along with the reason why we need sysinfo already in SPL.

File: recipes-kernel/linux/files/patches-6.1/0097-arm64-dts-ti-iot2050-Disable-lock-step-for-all-iot20.patch [hunk 5]
```diff
@@ -59,6 +59,3 @@ index be9c8db4c..87928ff28 100644
 -	/* lock-step mode not supported on this board */
 -	ti,cluster-mode = <0>;
 -};
```
Comment (jan-kiszka) [inline recipes-kernel/linux/files/patches-6.1/0097-arm64-dts-ti-iot2050-Disable-lock-step-for-all-iot20.patch:61]: This is unrelated noise, drop it.

Comment (jan-kiszka) [reply recipes-kernel/linux/files/patches-6.1/0097-arm64-dts-ti-iot2050-Disable-lock-step-for-all-iot20.patch:61]: Ah, sorry, it is a separate commit - that's fine.

File: recipes-app/iot2050-conf-webui/files/npm-shrinkwrap.json
Comment (huaqianli) [reply recipes-app/iot2050-conf-webui/files/npm-shrinkwrap.json:1]: Done

File: recipes-bsp/u-boot/files/0013-iot2050-add-smbios-and-sysinfo-support.patch
Comment (huaqianli) [reply recipes-bsp/u-boot/files/0013-iot2050-add-smbios-and-sysinfo-support.patch:1]: Removed the SMBIOS commit, and will submit it in another PR.

File: recipes-app/iot2050-conf-webui/files/src/components/ExtendedModules/SM1221_8DI/SM1221with8DIConf.js
Comment (huaqianli) [reply recipes-app/iot2050-conf-webui/files/src/components/ExtendedModules/SM1221_8DI/SM1221with8DIConf.js:1]: Done

File: recipes-app/iot2050-conf-webui/files/src/components/ExtendedModules/SM1223/SM1223_AC_DI8_DQ8RLY_Conf.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable camelcase */
+/* eslint-disable react/prop-types */
+import * as React from 'react';
+import SM1223Conf from './SM1223Conf';
+import uiString from '@/lib/uiString/SM1223.json';
```
Comment (BaochengSu) [inline recipes-app/iot2050-conf-webui/files/src/components/ExtendedModules/SM1223/SM1223_AC_DI8_DQ8RLY_Conf.js:5]: Imported but not used?

Comment (BaochengSu) [reply recipes-app/iot2050-conf-webui/files/src/components/ExtendedModules/SM1223/SM1223_AC_DI8_DQ8RLY_Conf.js:5]: Make sure no ESlint error during npm run build

Comment (huaqianli) [reply recipes-app/iot2050-conf-webui/files/src/components/ExtendedModules/SM1223/SM1223_AC_DI8_DQ8RLY_Conf.js]: Done

**Comment by BaochengSu** [comment]
LGTM

**Review by BaochengSu — APPROVED** [review]
